src/prefect_server/api/artifacts.py

```diff
@@ -0,0 +1,80 @@
+from prefect import models
+from prefect.utilities.plugins import register_api
+
+
+@register_api("artifacts.create_task_run_artifact")
+async def create_task_run_artifact(
+    task_run_id: str,
+    kind: str,
+    data: dict,
+    tenant_id: str = None,
+) -> str:
+    """
+    Create a task run artifact.
+
+    Args:
+        - task_run_id (str): the task run id
+        - kind (str): the artifact kind
+        - data (dict): the artifact data
+        - tenant_id (str, optional): the tenant id that this artifact belongs to. Defaults
+            to the tenant ID linked to the task run
+
+    Returns:
+        - str: the task run artifact ID
+    """
+    task_run = await models.TaskRun.where(id=task_run_id).first({"tenant_id"})
```

Comment:
It's important to check whether `task_run_id` is `None` before running this ORM call - Hasura treats `None` as "no where clause at all" so this would grab a random run and _possibly_ cause performance issues.

And just because it avoids a DB scan, I usually do:
```python
if not task_run_id: # handles empty strings as well, why not short-circuit
```